HW1/P4/P4.txt

```diff
@@ -1 +1,17 @@
 # Your discussion here
+Output:
+('CAPTAIN AMERICA', 'depth is:', 3, 'with # nodes:', 6408)
+('MISS THING/MARY', 'depth is:', 1, 'with # nodes:', 7)
+('ORWELL', 'depth is:', 1, 'with # nodes:', 9)
+
+1) What does this diameter imply about the maximum number of steps in which your search must be executed?
+
+The graph diameter is exactly what it’s name implies, it is the distance (or nodes) that we can traverse through the graph before we reach it’s maximum depth. The diameter implies the maximum number of steps to complete a depth full search from the parent node.		
+					
+2) What does it mean if a character does not have a defined distance during the search? 
+If a comic book character does not have a defined distance then he or she does not have many friends! To be more specific it means that he or she does not have any children in which they are connected to the parent node. This is completely possible when you have islanded nodes in a graph which are not connected and therefore isolated from a group of other nodes.
```

Comment:
Although it could also be that the source superhero has few friends (e.g. if a source super hero has no edges, then everyone else will have an undefined distance.)